pytorch_lightning/loops/batch/training_batch_loop.py

```diff
@@ -92,13 +91,13 @@ def run(self, batch: Any, batch_idx: int, dataloader_idx: int) -> AttributeDict:
             return AttributeDict(signal=-1)
 
         # hook
-        response = self.trainer.call_hook("on_train_batch_start", batch, batch_idx, dataloader_idx)
+        response = self.trainer.call_hook("on_train_batch_start", batch, batch_idx, 0)
```

Comment:
just curious, if it's 0 all the time in this hook, what's the point of even keeping it??

Reply:
backwards compatibility :) This has been part of the api for quite some time now, so probably there are a lot of people expecting this.

@carmocca do we have plans to deprecate the argument at all?

Reply:
> do we have plans to deprecate the argument at all?

I guess so, do you have any reason not to?